hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/TcpIpJoiner.java

```diff
@@ -73,6 +91,9 @@ public TcpIpJoiner(Node node) {
         }
         maxPortTryCount = tryCount;
         joinConfig = getActiveMemberNetworkConfig(config).getJoin();
+        previouslyJoinedMemberAddressRetentionDuration = TimeUnit.SECONDS.toMillis(
+                Integer.getInteger(PREVIOUSLY_JOINED_MEMBER_ADDRESS_RETENTION_DURATION_PROP,
```

Comment:
I moved this property into `ClusterProperty`.
For the granularity, we always use seconds granularit in all other properties (I don't see any example of `HOUR` granularity). So, I won't change this `SECONDS` granularity to avoid breaking convention.